fetch.bs

```diff
@@ -1150,6 +1209,72 @@ null), run these steps. <var>processBody</var> must be an algorithm accepting a
+an algorithm <var>processBodyError</var>, and an optional null,
+<a for=/>parallel queue</a>, or <a for=/>global object</a> <var>taskDestination</var> (default
+null), run these steps.
+<var>processBodyChunk</var> must be an algorithm accepting a <a for=/>byte sequence</a>.
+<var>processEndOfBody</var> must be an algorithm accepting no arguments. <var>processBodyError</var>
+must be an algorithm accepting an exception.
+<ol>
+ <li><p>Let <var>processAndReportEndOfBody</var> be an action the performs the following steps:
+ <ol>
+  <li><p><a for=response>Update timing info</a> for <var>response</var>.
+  <li><p>Call <var>processEndOfBody</var>.
+ </ol>
+
+ <li><p>Let <var>processAndReportBodyError</var> be an action that performs the following steps
+ given <a for=/>exception</a> <var>exception</var>:
+ <ol>
+  <li><p><a for=response>Update timing info</a> for <var>response</var>.
+  <li><p>Call <var>processBodyError</var> with <var>exception</var>.
+ </ol>
+
+ <li><p>Perform the <a for=body>incrementally read</a> algorithm given <var>response</var>'s
+ <a for=response>body</a>, <var>taskDestination</var>, <var>processBodyChunk</var>,
+ <var>processAndReportEndOfBody</var>, and <var>processAndReportBodyError</var>.
+</ol>
+
+<p>To <dfn export for=response>fully read</dfn> a <a for=/>responses</a> <var>response</var>,
+given an algorithm <var>processBody</var>, an algorithm <var>processBodyError</var>,
+and an optional null, <a for=/>parallel queue</a>, or <a for=/>global object</a>
+<var>taskDestination</var> (default null), run these steps. <var>processBody</var> must
+be an algorithm accepting a <a for=/>byte sequence</a>. <var>processBodyError</var> must be an
+algorithm accepting no arguments.
+<ol>
+ <li><p>Let <var>processAndReportBody</var> be an action that performs the following steps given
+ <a for=/>byte sequence</a> <var>data</var>:
+ <ol>
+  <li><p><a for=response>Update timing info</a> for <var>response</var>.
+  <li><p>Call <var>processBody</var> with <var>data</var>.
+ </ol>
+
+ <li><p>Let <var>processAndReportBodyError</var> be an action the performs the following steps:
+ <ol>
+  <li><p><a for=response>Update timing info</a> for <var>response</var>.
+  <li><p>Call <var>processBodyError</var>.
+ </ol>
+
+ <li><p>Perform the <a for=body>fully read</a> algorithm given <var>response</var>'s
+ <a for=response>body</a>, <var>taskDestination</var>, <var>processBodyChunk</var>,
+ <var>processAndReportBody</var>, and <var>processAndReportBodyError</var>.
+</ol>
+
+<p>To <dfn export lt="fully reading response as promise">fully read response body as promise</dfn>,
+given a <a for=/>response</a> <var>response</var>, run these steps:
+
+<ol>
+ <li><p>Let <var>promise</var> be the result of
+ <a for=/>fully reading response as promise</a> with <var>response</var>'s
+ <a for=response>body</a>.
+
+ <li><p><a for=/>Upon fulfillment</a> of <var>promise</var>, <a for=response>update timing info</a>
```

Comment:
I think this should be react as we also want to update these upon rejection, no?